Add a rudimentary test for @typedef

anandthakker opened the pull request:
@tmcw Added this in a super rudimentary format because I needed it for something. Couple of known issues I didn't have time to examine closely: 1) currently needs a `@name` tag, but presumably the name should be pulled off the `@typedef` tag (right?), and 2) if the typedef comment is at the end of a file, w/ no subsequent code node, it gets dropped.

tmcw commented:
:+1: ideally the PR for typedef is in a point release of doctrine before we go live with this

anandthakker commented:
> ideally the PR for typedef is in a point release of doctrine before we go live with this

Sounds good.

anandthakker commented:
@tmcw Just noticed you forked `doctrine` into this org.  Are you planning to have this module target the fork?

anandthakker commented:
Update: 0d36570ab7a5b2310ea12182a4969372b9612ad9 already took care of outputting properties in html.  Oddly, just rebased and reverted to previous doctrine release (0.6.4), and this basic @typedef example works just fine... possibly because of 42e8aee4780ccfd348492eac3fd3507c102e5a45.

anandthakker commented:
^ All of that means that all that's left here in this PR is a test case.

tmcw commented:
Yeah - it looks like @jfirebaugh already implemented this, or merged part of this branch?

jfirebaugh commented:
42e8aee4780ccfd348492eac3fd3507c102e5a45 was the only typedef related change I made.

anandthakker commented:
Yeah, I think it turned out that 42e8aee, along with actually looking for and rendering `@property` tags, was the only thing that was strictly needed.  (I believe upstream support for `@typedef` wasn't really necessary because, once @jfirebaugh fixed the name handling, doctrine's [default parsing](https://github.com/Constellation/doctrine/blob/master/lib/doctrine.js#L684) worked fine.)
